qa/rpc-tests/wallet_mergetoaddress_2.py

```diff
@@ -0,0 +1,348 @@
+from test_framework.util import assert_equal, initialize_chain_clean, \
+    start_node, connect_nodes_bi, sync_blocks, sync_mempools, \
+    wait_and_assert_operationid_status
+
+import time
+from decimal import Decimal
+from distutils.dir_util import copy_tree
+
+class WalletMergeToAddressTest (BitcoinTestFramework):
+
+    op_result_k = "op_result"
+    balance_delta_k = "balance_delta"
+    joinsplit_len_k = "joinsplit_len"
+    mergingUTXOs_k = "mergingUTXOs"
+    mergingNotes_k = "mergingNotes"
+    mergingTransparentValue_k = "mergingTransparentValue"
+    mergingShieldedValue_k = "mergingShieldedValue"
+
+    def import_data_to_data_dir(self):
+        #importing datadir resource
+        import os
+        os.chdir(os.path.dirname(__file__))
+        resource_file = os.getcwd() + "/resources/wallet_mergetoaddress_2/test_setup_.zip"
+        import zipfile
+        with zipfile.ZipFile(resource_file, 'r') as zip_ref:
+            zip_ref.extractall(self.options.tmpdir)
+        copy_tree(self.options.tmpdir + "/test_setup_", self.options.tmpdir)
+        shutil.rmtree(self.options.tmpdir + "/test_setup_")
+
+    def generate_notes(self, quantity):
+        myzaddr0 = self.nodes[0].z_getnewaddress()
+        results = []
+        for i in range(quantity):
+            self.nodes[0].generate(1)
+            self.sync_all()
+            self.nodes[1].generate(100)
+            self.sync_all()
+            results.append(self.nodes[0].z_shieldcoinbase("*", myzaddr0, 0))
+            wait_and_assert_operationid_status(self.nodes[0], results[i]['opid'])
+        self.nodes[1].generate(1)
+        self.sync_all()
+
+    def send_transparent(self, quantity):
+        for i in range(quantity):
+            self.nodes[1].sendmany("", {self.nodes[0].listaddresses()[0]: 10}, 100)
+        self.nodes[1].generate(100)
+        self.sync_all
+
+    def get_merged_transaction(self, unspent_transactions_before_merge, unspent_transactions_after_merge):
+        txid = unspent_transactions_after_merge[0]["txid"]
+        for aft in range(len(unspent_transactions_after_merge)):
+            for bef in range(len(unspent_transactions_before_merge)):
+                if (unspent_transactions_after_merge[aft]["txid"] == unspent_transactions_before_merge[bef]["txid"]):
+                    break
+                if (bef == len(unspent_transactions_before_merge) - 1):
+                    txid = unspent_transactions_after_merge[aft]["txid"]
+        return self.nodes[2].gettransaction(txid)
+
+    def z_mergetoaddress_t2t(self, utxos_quantity):
+        mytaddr2 = self.nodes[2].listaddresses()[0]
+        listunspent_before_merge = self.nodes[2].listunspent(0)
+        balance_before_merge = self.nodes[2].z_getbalance(mytaddr2)
+        result = self.nodes[0].z_mergetoaddress(["ANY_TADDR"], mytaddr2, 0, utxos_quantity, 0)        
+        wait_and_assert_operationid_status(self.nodes[0], result['opid'], "success", "", 3600)
+        self.nodes[1].generate(100)
```

Comment:
I don't know if mining 100 blocks is really a necessity. If it is done to prevent some timing issues, maybe it is possible to main blocks (i.e. mine a block and check the condition) until the issue is solved.

Reply:
Moved to single block mining on node1 (apart from `generate_notes` method).